### PR TITLE
Input Placeholder i18n support

### PR DESCRIPTION
**Pull request by shinxi**
From https://github.com/predixdesignsystem/px-datetime-picker/issues/38
* Add localization support to input placeholder.
* Add demo and test case.

@joshsylvester @evanjd @sheela-svmx @benjaminliugang pls help review. 
I haven't updated the version number due to previous discussion https://github.com/predixdesignsystem/px-spinner/pull/13#discussion_r261537745, @evanjd feel free to let me know if want me update the version as well in this PR.

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -127,6 +128,12 @@
      */
     demoProps: {
 
+      resources: {
+        type: Array,
+        defaultValue: { en: { YY: "Year", MM: "Month", DD: "Day" } },
```
Comment (joshsylvester) [inline demo/index.html]: to be consistent with the current demo, wouldn't this need to be: 

`defaultValue: { en: { YYYY: "Year", MM: "Month", DD: "Day" } },`

Comment (shinxi) [reply demo/index.html]: @joshsylvester the date format in current demo **is** `MM/DD/YY`...
https://github.com/predixdesignsystem/px-datetime-common/blob/fe03eafd24f65442ddbf1c842ed60f4f4a62aad9/demo/index.html#L116

**Comment by shinxi** [comment]
@joshsylvester @evanjd any update?

File: demo/index.html [hunk 2]
```diff
@@ -127,6 +128,12 @@
      */
     demoProps: {
 
+      resources: {
+        type: Array,
```
Comment (evanjd) [inline demo/index.html]:
```suggestion
        type: Object,
```

**Comment by shinxi** [comment]
> Updating the resources property does not regenerate the placeholder text, making the demo for this prop non-functional as far as I can tell.

@evanjd Good point, I just replaced with a read only text prop, see below.
image
Feel free to let me know if it's ok or any better idea?

**Comment by evanjd** [comment]
@shinxi Ideally the fix would be to make the component responsive to `resources` prop changes.

**Comment by shinxi** [comment]
@evanjd this was also my first idea. But after trying out, the demo component won't be responsive to `resources` prop change until change the moment format.
E.g., Change `YY: Year` to `YY: Year2`, it won't take effect until change the moment format `MM/DD/YY` to `MM/DD/YYYY`, and change back to `MM/DD/YY`.
That's because the `placeholder` is listening on the `momentFormat`, right?
image
Given this is only for demo purpose and we won't change resources in real user scenario, marking this resources readonly in the demo sounds ok to me. @evanjd What do you think?

**Comment by evanjd** [comment]
It's not so much the demo I'm concerned with, it's the confusion it may create for devs using this when the resources prop doesn't take if added or updated after the component is attached.

What about adding `resources` as a parameter for the placeholder computed prop?

**Comment by shinxi** [comment]
> What about adding resources as a parameter for the placeholder computed prop?

Updated, pls help review again.

**Comment by shinxi** [comment]
@evanjd Pls help review.

**Comment by shinxi** [comment]
> One last problem, the inputs do not resize when the placeholder is changed.

For real use case, don't think user will change locale once page is loaded, but anyway, good catch.
> Firing the _formatChanged observer (in px-datetime-entry) when the resources prop is set should fix it.

Actually calling `_sizeInputs` should do the work and more efficient and accurate than calling `_formatChanged`. 

Removed `resources` parameter for placeholder computed prop, and instead, add `_onResourcesUpdated(resources)` observer to update placeholder text and resize input when resources is changed.
@evanjd pls help review again.

**Comment by shinxi** [comment]
@evanjd pls help...

**Comment by shinxi** [comment]
> In future can you avoid amending and force pushing? It makes it hard to review what you've changed. 

I think you can click the [force-pushed](https://github.com/predixdesignsystem/px-datetime-common/compare/cbd668661bebbf6915ae090df63504038aeec064..303d175b8269b3dc518d8301d5a922bb96f35274) link to review the new changes, right?
image
Our team has some rule that one PR can only have one commit to keep the history clean. Anyway, next time I can change(add more commits) for sure if you want.

**Comment by evanjd** [comment]
There’s no diff. I want to see what’s changed in response to review, so I don’t have to read the whole change again. 

There’s no need to keep the revision history clean in response to PR comments, all commits are squashed in the merge commit.

**Comment by shinxi** [comment]
> There’s no diff. I want to see what’s changed in response to review, so I don’t have to read the whole change again.

Hmm...I can see the diff clearly via the above [force-pushed](https://github.com/predixdesignsystem/px-datetime-common/compare/cbd668661bebbf6915ae090df63504038aeec064..303d175b8269b3dc518d8301d5a922bb96f35274) link.
image

**Comment by evanjd** [comment]
Ah, didn't know it was under the force push link, thanks.

Still, my comment stands. Don't force push if you don't need to. The history gets squashed on merge, there's no need to fuss over the revision history.

**Comment by shinxi** [comment]
> Still, my comment stands. Don't force push if you don't need to. The history gets squashed on merge, there's no need to fuss over the revision history.

Sure, just different team standards, next time I will follow yours.

**Pull request merged**